service/src/test/java/skills/intTests/AdminEditSpecs.groovy

```diff
@@ -1095,4 +1095,50 @@ class AdminEditSpecs extends DefaultIntSpec {
         u2LevelPostDelete == 1
         u2SubjLevelPostDelete == 1
     }
+
+    def "deleting a subject should remove orphaned points and achievements"() {
+        def project = SkillsFactory.createProject()
+        def subject = SkillsFactory.createSubject()
+        def subject2 = SkillsFactory.createSubject(1, 2)
+        def skill1 = SkillsFactory.createSkill(1, 1, 1, 0, 5)
+        skill1.pointIncrement = 50
+        def skill2 = SkillsFactory.createSkill(1, 2, 2, 0, 5)
+        skill2.pointIncrement = 20
+
+        skillsService.createProject(project)
+        skillsService.createSubject(subject)
+        skillsService.createSubject(subject2)
+        skillsService.createSkill(skill1)
+        skillsService.createSkill(skill2)
+
+        def users = getRandomUsers(2)
+        def user1 = users[0]
+        def user2 = users[1]
+
+        when:
+        skillsService.addSkill(skill1, user1, new Date().minus(5))
+        skillsService.addSkill(skill1, user1, new Date().minus(1))
+        skillsService.addSkill(skill1, user2, new Date())
+        skillsService.addSkill(skill2, user2, new Date())
+
+        def u1Level = skillsService.getUserLevel(project.projectId, user1)
+        assert u1Level == 2
+        def u2Level = skillsService.getUserLevel(project.projectId, user2)
+        assert u2Level == 1
+
+        def projectUsers = skillsService.getProjectUsers(project.projectId)
+        assert projectUsers.data.find { it.userId == user1 && it.totalPoints == 100 }
+        assert projectUsers.data.find { it.userId == user2 && it.totalPoints == 70 }
+
+        skillsService.deleteSubject(subject)
+        def projectUsersPostDelete = skillsService.getProjectUsers(project.projectId)
+        def u1LevelPostDelete = skillsService.getUserLevel(project.projectId, user1)
+        def u2LevelPostDelete = skillsService.getUserLevel(project.projectId, user2)
+
+        then:
+        !projectUsersPostDelete.data.find { it.userId == user1 }
+        projectUsersPostDelete.data.find { it.userId == user2 && it.totalPoints == 20 }
+        !u1LevelPostDelete
+        u2LevelPostDelete == 1
```

Comment:
please add extra validation to make sure that skill's and subject's user_points are removed  as well as skill/subj/proj user_achievements are gone too